updates to procedure introduction (#5572)
SDU-eScience/UCloud: provider-integration/im2/docs/src/procedures/procedures_intro.md

### Initial testing

Eventually code will reach a functional stage. At this point the issue becomes "ready for initial testing". In this

> stage, the code is tested by the assigned developer.  Code can be tested both manually and automatically. 

Reviewer: ```suggestion
stage, the code is tested by the assigned developer. Code can be tested both manually and automatically. 
```

> Manual testing is first performed on a local created UCloud environment. Once the initial manual test are satisfactory.

Reviewer: ```suggestion
Manual testing is first performed on a locally created UCloud environment. Once the initial manual test are satisfactory.
```

The manual testing moves onto the development system. The development system contains a 
software and hardware stack similar to the one used in the production environment. This allows us to more accurately 
test code. The development system is commonly used in our system to act as a staging environment. Our experience has 
shown that with our team size, a separate staging environment did not provide sufficient benefits.

> Automatic tests are conducted when a PR is merged into the `master` branch consisting of a E2E test assuring that everything still

Reviewer: ```suggestion
Automatic tests are conducted when a PR is merged into the `master` branch consisting of an E2E test suite, assuring that everything still
```

> works as intended. The automatic testing is run using GitHub Actions and creates an entire local cloud system to perform

Reviewer: ```suggestion
works as intended. The automatic testing runs using GitHub Actions and creates an entire local cloud system to perform
```

## Monitoring and Auditing

> Audit logs are consumed using our own audit ingestion service along side a postgres database and can be presented and 

Reviewer: ```suggestion
Audit logs are consumed using our own audit ingestion service along side a Postgres database and can be presented and 
```
